src/test/java/com/example/demo/services/AccountServiceTests.java

```diff
@@ -0,0 +1,155 @@
+package com.example.demo.services;
+
+import static org.easymock.EasyMock.*;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import org.easymock.EasyMockRunner;
+import org.easymock.Mock;
+import org.easymock.TestSubject;
+import org.junit.Test;
+import org.junit.runner.*;
+
+import com.example.demo.entities.User;
+import com.example.demo.exceptions.BadCredentialsException;
+import com.example.demo.exceptions.ExistingUserException;
+import com.example.demo.exceptions.NonExistentUserException;
+import com.example.demo.models.LoginModel;
+import com.example.demo.models.SignupModel;
+import com.example.demo.repositories.UsersRepository;
+import com.example.demo.utils.IHashUtil;
+import com.example.demo.utils.IJwtUtil;
+
+@RunWith(EasyMockRunner.class)
+public class AccountServiceTests {
+	
+	@Mock
+	IJwtUtil jwtUtilMock;
+	@Mock
+	IHashUtil hashUtilMock;
+	@Mock
+	UsersRepository usersRepoMock;
+	
+	@TestSubject
+	AccountService accountControler=new AccountService(hashUtilMock,jwtUtilMock,usersRepoMock);
+	
+	//login tests
+	@Test
+	public void loginValidCredentialsShouldReturnMockedToken() throws Exception{
+		LoginModel loginModel=new LoginModel();
+		loginModel.email="rajko@mail.com";
+		loginModel.password="rajko123";
+		
+		User userEntity=new User();
+		userEntity.setEmail(loginModel.email);
+		userEntity.setPasswd(loginModel.password);
+		
+		List<User> users=new ArrayList<>();
+		users.add(userEntity);
+		
+		expect(hashUtilMock.checkPassword(loginModel.password,userEntity.getPasswd())).andReturn(true);
+		expect(usersRepoMock.findByEmail(loginModel.email)).andReturn(users);
+		expect(jwtUtilMock.generateToken(userEntity,new HashMap<String, Object>())).andReturn("fakeJWT");
+		replay(hashUtilMock,usersRepoMock,jwtUtilMock);
```

Comment:
we can extend Test class with EasyMockSupport use replayAll() for this line, and after test method is called we need to call verifyAll() so that we have complete unit tests